File: code/kauthenticity/week3/prog_64062.js
function solution(stones, k) {
  var answer = 0;
  let left = stones.reduce(
    (minVal, cur) => (minVal < cur ? minVal : cur),
    stones[0]
  );
  let right = stones.reduce(
    (maxVal, cur) => (maxVal > cur ? maxVal : cur),
    stones[0]
  );

  while (left <= right) {
    const mid = Math.floor((left + right) / 2); // 건널 수 있는 사람 수

    let zeroDist = 0;
    for (let i = 0; i < stones.length; i++) {
      if (stones[i] - mid <= 0) {
        zeroDist++;
      } else {
        zeroDist = 0;
      }

      if (zeroDist === k) {
        break;
      }
    }

    if (zeroDist === k) {
      right = mid - 1;
    } else {
      left = mid + 1;
    }
  }
  answer = left;

  return answer;
}

File: code/jonghyeonjo99/week3/prog_64062.py
def solution(stones, k):
    answer = 0
    left = 1
    right = max(stones)

    while(left <= right):
        cnt = 0
        results = []
        mid = (left + right) // 2

        for i in range(len(stones)):
            if stones[i] <= mid:
                cnt += 1
            else:
                results.append(cnt)
                cnt = 0
                
        results.append(cnt)
        result = max(results)

        if result < k:
            left = mid + 1

        else:
            answer = mid
            right = mid -1
    return answer